src/v/features/enterprise_feature_messages.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "ssx/sformat.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <fmt/core.h>
+
+#include <string>
+#include <string_view>
+
+namespace features::enterprise_error_message {
+
+constexpr std::string_view required
+  = "A Redpanda Enterprise Edition license is required";
+
+constexpr std::string_view request_or_trial
+  = "To request an Enterprise license, please visit "
+    "https://redpanda.com/upgrade. To try Redpanda Enterprise for 30 days, "
+    "visit https://redpanda.com/try-enterprise.";
+
+constexpr std::string_view get_started
+  = "For more information, see "
+    "<https://docs.redpanda.com/current/get-started/licenses>.";
+
+inline ss::sstring license_nag(const auto& features) {
+    return ssx::sformat(
+      fmt::runtime(
+        "{} to use the currently enabled feature(s): ({}). Please provide a "
+        "valid license key via rpk using 'rpk cluster license set <key>', or "
+        "via Redpanda Console. {} {}}"),
+      required,
+      fmt::join(features, ", "),
+      request_or_trial,
+      get_started);
+}
+
+inline constexpr std::string upgrade_failure(const auto& features) {
+    return fmt::format(
+      "{} to use the currently enabled feature(s): ({}). To apply your "
+      "license, downgrade this broker to the pre-upgrade version and provide a "
+      "valid license key via rpk using 'rpk cluster license set <key>', or via "
+      "Redpanda Console. {} {}",
+      required,
+      fmt::join(features, ", "),
+      request_or_trial,
+      get_started);
+}
+
+inline ss::sstring cluster_property(std::string_view name, auto value) {
+    return ssx::sformat(
+      fmt::runtime(
+        "Rejected {}: '{}' '{}' is restricted to clusters with an Enterprise "
```

Comment:
nitpick: maybe (just maybe):

```suggestion
        "Rejected {}: '{}' - '{}' is restricted to clusters with an Enterprise "
```

Reply:
That's not what the requirements say. @mattschumpert ?

Reply:
Oh, there are reqs...ignore me!